loader/src/Loader.cc

```diff
@@ -431,7 +437,8 @@ namespace gz
 
       // NOTE: We open using RTLD_LOCAL instead of RTLD_GLOBAL to prevent the
       // symbols of different libraries from writing over each other.
-      void *dlHandle = dlopen(_full_path.c_str(), RTLD_LAZY | RTLD_LOCAL);
+      int dlopenMode = RTLD_LAZY | RTLD_LOCAL | (_noDelete ? RTLD_NODELETE : 0);
```

Comment:
we might need an `ifdef` for `RTLD_NODELETE` on windows

Reply:
Done in https://github.com/gazebosim/gz-plugin/pull/102/commits/644f61377667b53e6a554763590ad97d26387374

Reply:
> Apparently, the behavior of dlopen is different between Linux and macOS. On Linux, if you dlopen (with RTLD_NODELETE) , dlclose, and dlopen again, since the plugin is not deleted, dlopen will reuse the existing plugin. On macOS, however, even though the plugin is not deleted, dlopen will treat the second dlopen as though a new library is being loaded and end up loading it into a new memory location. Thus, the type of test I was using won't work. Since we're running out of time, I've decided to enable the test only on Linux. If there are other ways to test this, I'm open to ideas.

Should we add `__APPLE__` to the `ifdef`?